### DISCUSSION TITLE
Best ways to backup OpenTracks

### DISCUSSION THREAD

**Discussion by edern-appere** [OpenTracksApp/OpenTracks#1277] [Q&A] [answered]
Hi everyone !

Recently I had to make a fresh new install of my phone. Before that, I exported all my tracks in GPX files (thanks to the auto-export feature !). Once I imported them back into my freshly installed OpenTracks, I noticed that all the tracks where tagged as cycling although, some where hiking, some ski and most of them cycling. I'm trying to figure out which tracks belongs to which activity. I have a question : is this a bug or is this due to the GPX file format I used for export ? I'm not familiar with KMZ file, but if this could resolve this issue, I can use that.

As a side note, I noticed that there is no feature to  export OpenTracks settings. Setting-up OpenTracks is quite quick and easy but it would be great to have an export feature, especially for the customized recording dashboards. Is this already implemented and I have missed it ?

Thanks for your help !

**Comment by dennisguse** [comment, marked as answer]
Welcome back to the party :partying_face: 

1. Backups
Use either KMZ or KML, because GPX cannot contain all the information.
Regarding the actual activity: it is exported as a localized string (in your language) if you import on another device with a different language, everything will be imported as unknown.
There is no documentation for this; definitely a point to improve.
It is only in code.
PS/ I have been writing this just from my memory.
Relevant code is in GPXTrackImporter and KMLTrackImporter.
2. We cannot export the settings (at the moment).
So, you didn't miss anything.
Also a nice feature...

**Reply by edern-appere** [reply]
Thanks for the answers ! 
I'll switch to KML files then, because my Nextcloud app to read and display files seems to allow only GPX or KML files.

**Reply by edern-appere** [reply]
By the way, you said "Welcome back to the party". Even though I did not post many issues or discussions here, I never stopped using OpenTracks. It's my one and only essential app when I go biking ! And from what I read here, I'm not the only one ! OpenTracks has become the best bicycle computer replacement app. You can be proud of your work @dennisguse !

**Reply by dennisguse** [reply]
Doing it for the fun :sunglasses:

**Comment by Zugschlus** [comment]
Hi, how would I restore a file written by Opentracks _in_ Opentracks? I have written all my tracks to a single file by choosing "Export all (one file for all tracks)" and moved that file to the new phone by using nearby share. It now sits in the Download folder. Opentracks "import all" says  it cannot use this folder. I have copied the file to a new "opentracks" folder, pointed Opentracks to that, folder and clicked "use this folder". Opewntracks says "importing from opentracks, completed with one error", and show errors says "Parser error, cannot import track without any locations". How do I proceed?
Exporting one file per track is a nuisance since the files are transferred (and placed in the download folder) one by one, and I have nearly a thousand tracks.

**Reply by dennisguse** [reply]
Are any tracks successfully imported?
Might be that only one track has no data. 
Or is nothing imported at all?

**Reply by dennisguse** [reply]
Which file format?

**Comment by tinogithub** [comment]
Hello,
I haven't exported my data manually but did a complete backup of my phone before I installed a new OS. Is there a way to get my tracks back again?

**Reply by pstorch** [reply]
What kind of backup? Are able to look into it for files or restore it?

**Reply by tinogithub** [reply]
It's a copy of the whole file structure that you get access to via USB. With the directories Android/Download/Movies/ etc.

**Reply by pstorch** [reply]
I think the OpenTracks DB is in an internal storage not accessible via USB (at least without root privileges).

**Reply by tinogithub** [reply]
Ok. That's unfortunate. As it makes doing backups more complicated. But I know that is an overall problem with apps. I have found no other way than opening every app and doing their individual backup method. Unfortunately I have forgotten OpenTracks.

**Reply by dennisguse** [reply]
The file you are looking for would be in `/data/data/de.dennisguse.opentracks/databases/database.db`.
Images are also somewhere there. 

However, this requires a complete backup (like Titanium or other tools requiring root or run as recovery image).
